### ISSUE TITLE
F-Droid build failed

### ISSUE THREAD

**Issue by linsui** [Catfriend1/syncthing-android#891]
https://monitor.f-droid.org/builds/log/com.github.catfriend1.syncthingandroid/1200100#site-footer

Could you please take a look? Thanks!

**Comment by bt90** [comment]
Is fdroid release currently maintained or should we switch to the playstore?

**Comment by xeruf** [comment]
It seems that switching is not as easy, as the play store release has a different signature...

**Comment by Mrnofish** [comment]
> the play store release has a different signature...

Correct, you'd have to export the settings from the F-Droid version and import them in whatever version you decide to use: [the most recent one can be downloaded from Github](https://github.com/Catfriend1/syncthing-android/releases/tag/v1.20.1.0), for those who would rather not use Google's Play Store.

It does seem that there has been no work being done on the fork in the last month and it was very sparse before, however this means no updates to the core Syncthing application this one is a front-end to.

Another alternative, and one that may become obligated if work doesn't resume, is switching back to [the original Syncthing for Android, also on F-Droid](https://f-droid.org/en/packages/com.nutomic.syncthingandroid/).

**Comment by xeruf** [comment]
Nice, I wasn't sure whether I could trust the Import/Export function, but it worked fine :)

**Comment by Catfriend1** [comment]
Hi, 

I'll take a look as soon as time permits me to do it. I noticed the build problem, probably related to the NDK, but at the moment, I do only have a Windows builder on my PC. Will require me to install a fresh debian linux and check out how and why it fails.

**Comment by bt90** [comment]
java.io.FileNotFoundException: https://dl.google.com/android/repository/sys-img/android-desktop/sys-img2-3.xml

**Comment by Catfriend1** [comment]
Where is this referenced?

**Comment by bt90** [comment]
Part of the original log posted by OP.

**Comment by otbutz** [comment]
This seems to be the real problem:

```
/home/vagrant/build/com.github.catfriend1.syncthingandroid/syncthing/../../syncthing-android-prereq/android-ndk-r24/toolchains/llvm/prebuilt/linux-x86_64/bin/clang: line 1: /home/vagrant/build/com.github.catfriend1.syncthingandroid/syncthing/../../syncthing-android-prereq/android-ndk-r24/toolchains/llvm/prebuilt/linux-x86_64/bin/clang-12: No such file or directory
```

**Comment by Catfriend1** [comment]
> This seems to be the real problem:
> 
> ```
> /home/vagrant/build/com.github.catfriend1.syncthingandroid/syncthing/../../syncthing-android-prereq/android-ndk-r24/toolchains/llvm/prebuilt/linux-x86_64/bin/clang: line 1: /home/vagrant/build/com.github.catfriend1.syncthingandroid/syncthing/../../syncthing-android-prereq/android-ndk-r24/toolchains/llvm/prebuilt/linux-x86_64/bin/clang-12: No such file or directory
> ```

Right, we had this problem since NDK23 as it was a bug in the NDK which I fixed using a workaround. So I need to figure out what Google changed here again. Thank you.:)

**Comment by Catfriend1** [comment]
NDK r24: The problem is the file "/root/work/syncthing-android-prereq/android-ndk-r24/toolchains/llvm/prebuilt/linux-x86_64/bin/clang" which has "clang-14" as content. It needs to be:
```
`dirname $0`/clang-14 "$@"
```

**Comment by Catfriend1** [comment]
New release tags created:
- https://github.com/Catfriend1/syncthing-android-fdroid/releases/tag/v1.20.1.0
- https://github.com/Catfriend1/syncthing-android-fdroid/releases/tag/v1.20.3.0
-

**Comment by Mrnofish** [comment]
Thank you very much!

**Comment by bt90** [comment]
The current build finished successfully. The new release should be offered by f-droid soon.
